src/main/java/nl/hsac/fitnesse/fixture/util/SoapSigningHelper.java

```diff
@@ -0,0 +1,248 @@
+package nl.hsac.fitnesse.fixture.util;
+
+import nl.hsac.fitnesse.fixture.Environment;
+import nl.hsac.fitnesse.fixture.slim.SlimFixtureException;
+import org.apache.wss4j.common.crypto.Crypto;
+import org.apache.wss4j.common.crypto.CryptoFactory;
+import org.apache.wss4j.common.ext.WSSecurityException;
+import org.apache.wss4j.dom.WSConstants;
+import org.apache.wss4j.dom.message.WSSecHeader;
+import org.apache.wss4j.dom.message.WSSecSignature;
+import org.apache.wss4j.dom.message.WSSecTimestamp;
+import org.apache.wss4j.dom.message.WSSecUsernameToken;
+import org.w3c.dom.Document;
+
+import javax.xml.crypto.dsig.CanonicalizationMethod;
+import javax.xml.crypto.dsig.DigestMethod;
+import javax.xml.soap.*;
+import java.io.ByteArrayInputStream;
+import java.util.Arrays;
+import java.util.Properties;
+
+public class SoapSigningHelper {
+
+    public static final String VALID_DIGEST_ALGORITHMS = "SHA1, SHA256, SHA512, RIPEMD160";
```

Comment:
Why not initialise these static fields with arrays (or lists), instead of splitting on each usage?

Reply:
I knew this question was coming :-)
I use these constants in the javadoc comments, so that the context helper documentaion can show the possibilities (and needs no maintenance when the lists of supported options expands). As the only way to reference constants from javadoc is when the constants are immutable primitives or Strings. When using an array or list, the array itself is immutable, but the values inside are not and thus they are not shown in the javadoc output.

Reply:
Quite a good reason ;-)
I don't believe you use the splits more than once in normal behaviour, so leave them as is, but maybe a comment to indicate why this is done?